test/api/fake_api.dart

```diff
@@ -29,18 +30,18 @@ class _PreparedSuccess extends _PreparedResponse {
 /// An [http.Client] that accepts and replays canned responses, for testing.
 class FakeHttpClient extends http.BaseClient {
 
-  http.BaseRequest? lastRequest;
+  Iterable<http.BaseRequest> get requestHistory => _requestHistory;
+  List<http.BaseRequest> _requestHistory = [];
 
-  http.BaseRequest? takeLastRequest() {
-    final result = lastRequest;
-    lastRequest = null;
+  List<http.BaseRequest> takeRequests() {
```

Comment:
How about also making the return type `Iterable` here?

Reply:
I didn't apply the same refactor there because `takeRequests` do not need to protect the list from modification. So `List` would be an accurate return type.